blockdag/dag.go

```diff
@@ -1630,30 +1630,83 @@ func (dag *BlockDAG) blueBlocksBetween(lowHash, highHash *daghash.Hash, maxEntri
 	// first iterate on the selected parent chain of the highNode
 	// until we find a new highNode
 	// where highNode.blueScore-lowNode.blueScore+1 <= maxEntries
-
+	//
+	// We also keep track of highNode's selected parent chain child
```

Comment:
In the beginning of this comment we state "In order to get no more then maxEntries of blue blocks from... ".
I believe we are no longer only collecting blue blocks.

It might be also a good idea to state the assumption that most blocks are blue, and therefore `highNode.blueScore-lowNode.blueScore+1` is a good estimation to total amount of blocks we are going to get.

Also it could be nice to have an explanation why moving the highNode one notch lower is a good idea.